### PR TITLE
GLUPAY-5835: Update metrics-datadog library

### PR DESCRIPTION
**Pull request by pdconant**
Dropwizard 1.3.4 → 3.0.2
Dropwizard-metrics 4.0.2 → 4.2.19
Jackson 2.9.6 → 2.15.2
Apache HTTP client 4.5.5 → 5.2.1
dogstatsd-client 2.6.1 → 4.2.0
Junit 4.12 → 5.10.0
Mockito 1.10.19 → 5.5.0
Slf4j 1.7.25 → 2.0.9
Various Maven plugin updates

### CODE REVIEW

File: metrics-datadog/src/test/java/org/coursera/metrics/datadog/TaggedNameTest.java [hunk 1]
```diff
@@ -1,10 +1,10 @@
 package org.coursera.metrics.datadog;
 
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
 
 import static org.coursera.metrics.datadog.TaggedName.TaggedNameBuilder;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (jonathan-taub) [inline metrics-datadog/src/test/java/org/coursera/metrics/datadog/TaggedNameTest.java]: I thought we are "against" asterisk imports.

File: metrics-datadog/src/test/java/org/coursera/metrics/datadog/transport/UdpTransportTest.java [hunk 2]
```diff
@@ -6,19 +6,20 @@
...
 import com.alibaba.dcm.DnsCacheManipulator;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
 
-import org.junit.Before;
-import org.junit.Test;
-
-import static org.junit.Assert.*;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (jonathan-taub) [inline metrics-datadog/src/test/java/org/coursera/metrics/datadog/transport/UdpTransportTest.java:13]: Same as above.

**Review by jonathan-taub — APPROVED** [review]
Made a couple of minor comments.

File: metrics-datadog/src/test/java/org/coursera/metrics/datadog/TaggedNameTest.java (hunk 1 above)
Comment (pdconant) [reply metrics-datadog/src/test/java/org/coursera/metrics/datadog/TaggedNameTest.java]: Oops -- my intention was to keep wildcard imports where they were, but not expand their use. I didn't intend to change the coding standards for this project. I'll replace this wildcard import but leave the other.

**Pull request merged**